multi-node/aws/pkg/cluster/cluster.go

```diff
@@ -42,38 +45,132 @@ func New(cfg *config.Cluster, awsDebug bool) *Cluster {
 	}
 
 	return &Cluster{
-		clusterName: aws.String(cfg.ClusterName),
-		svc:         cloudformation.New(session.New(awsConfig)),
+		Cluster: *cfg,
+		session: session.New(awsConfig),
 	}
 }
 
 type Cluster struct {
-	clusterName *string
-	svc         *cloudformation.CloudFormation
+	config.Cluster
+	session *session.Session
 }
 
 func (c *Cluster) ValidateStack(stackBody string) (string, error) {
-	input := &cloudformation.ValidateTemplateInput{
+	validateInput := cloudformation.ValidateTemplateInput{
 		TemplateBody: &stackBody,
 	}
 
-	validationReport, err := c.svc.ValidateTemplate(input)
+	cfSvc := cloudformation.New(c.session)
+	validationReport, err := cfSvc.ValidateTemplate(&validateInput)
 	if err != nil {
-		return "", fmt.Errorf("Invalid cloudformation stack: %v", err)
+		return "", fmt.Errorf("invalid cloudformation stack: %v", err)
 	}
 
-	return validationReport.String(), err
+	describeStacksInput := cloudformation.DescribeStacksInput{
+		StackName: aws.String(c.ClusterName),
+	}
+
+	//Find out if stack exists already. This determines whether we should do subnet conflict validatio
+	var stackExists bool
+	stackNotExistExpr := regexp.MustCompile(fmt.Sprintf("^ValidationError: Stack with id %s does not exist", c.ClusterName))
+
+	describeStacksOutput, err := cfSvc.DescribeStacks(&describeStacksInput)
+	if err != nil {
+		if stackNotExistExpr.Match([]byte(err.Error())) {
+			//No results for a list operation is not an error!!! (unless your AWS)
+			stackExists = false
+		} else {
+			return "", fmt.Errorf("error describing stack: %v", err)
+		}
+	} else {
+		stackExists = true
+		if len(describeStacksOutput.Stacks) > 1 {
+			return "", fmt.Errorf("expected error: found more the one stack with name %s", c.ClusterName)
+		}
+	}
+
+	//if the stack already exists, is doesn't make sense to validate the existing VPC for subnet conflicts
+	if c.VPCID != "" && !stackExists {
+		fmt.Println("Existing VPC detected. Will validate for subnet cidr conflicts")
+		if err := c.validateExistingVPC(); err != nil {
+			return "", err
+		}
+	}
+
+	return validationReport.String(), nil
+}
+
+func (c *Cluster) validateExistingVPC() error {
```

Comment:
Would help to add a comment for this function describing it's intended functionality.